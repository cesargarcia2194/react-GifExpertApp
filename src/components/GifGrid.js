import React from 'react'
import { GifGrdItem } from './GifGrdItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({category}) => {
    
    const {data:images, loading} = useFetchGifs(category);
    return (
        <>
            <h3 className="animate__animated animate__fadeInDown">{category}</h3>
            {loading && <p>Loading...</p>}
            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGrdItem 
                            key= {img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}
