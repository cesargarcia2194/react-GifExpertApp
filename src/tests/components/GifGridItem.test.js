import {shallow} from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en el componente GifGridItem', () => {
    const title = 'Un titulo';
    const url = 'http://localhost:3000/products';
    const wrapper = shallow(<GifGridItem url={url} title= {title} />)
    test('Debe mostrar el componente <GifGrdItem /> correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de tener un parrafo con title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title)
    });

    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('debe contoner la clase animate__fadeIn', () => {
        const div = wrapper.find('div');
        //console.log(div.prop('className').includes('animate__fadeIn'))
        expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
    })
    
})