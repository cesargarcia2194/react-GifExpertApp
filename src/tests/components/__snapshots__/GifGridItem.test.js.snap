// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGridItem Debe mostrar el componente <GifGrdItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Un titulo"
    src="http://localhost:3000/products"
  />
  <p>
    Un titulo
  </p>
</div>
`;
